apps/src/music/views/TimelineSimple2Events.jsx

```diff
@@ -24,6 +25,41 @@ const TimelineSimple2Events = ({
     return currentUniqueSounds.indexOf(id);
   };
 
+  function getFunctionBounds(orderedFunction) {
+    let left = Number.MAX_SAFE_INTEGER,
+      top = Number.MAX_SAFE_INTEGER,
+      right = 0,
+      bottom = 0;
+
+    orderedFunction.playbackEvents.forEach(playbackEvent => {
+      left = Math.min(left, playbackEvent.when);
+      right = Math.max(right, playbackEvent.when + playbackEvent.length);
```

Comment:
Can we use `startMeasure` and `endMeasure` on `FunctionEvents` for the left and right values here? In theory these should be calculated correctly since we update the endMeasure whenever we exit a function context (aside from when_run, but looks like we ignore that outermost function as it is).

Additionally, I think we might be able to just loop through `orderedFunctions` to construct the timeline elements, rather than constructing the function bounds from `orderedFunctions` and using `playbackEvents` for the timeline elements. We'd probably still need to loop through each playback event to find the top and bottom bounds, but using just `orderedFunctions` as a source of truth should hopefully help optimize this.

Reply:
The tricky part for the vertical aspect of a function is that we must make sure we encompass play events generated by functions that it calls, recursively.

Reply:
Ah yep that's true. Discussed offline, but this approach should be fine for now, and we can revisit as we tweak the timeline UI.

Reply:
Agreed.  With the structure we now have, it would be great to try some different approaches to this timeline to see what we can come up with.